go-controller/pkg/node/openflow_manager.go

```diff
@@ -275,12 +275,12 @@ func checkPorts(netConfigs []*bridgeUDNConfiguration, physIntf, ofPortPhys strin
 
 		}
 		if netConfig.ofPortPatch != curOfportPatch {
-			if netConfig.isDefaultNetwork() || curOfportPatch != "" {
+			if netConfig.isDefaultNetwork() {
 				klog.Errorf("Fatal error: patch port %s ofport changed from %s to %s",
 					netConfig.patchPort, netConfig.ofPortPatch, curOfportPatch)
 				os.Exit(1)
 			} else {
-				klog.Warningf("Patch port %s removed for existing network", netConfig.patchPort)
+				klog.Warningf("UDN patch port %s changed for existing network from %v to %v. Expecting bridge config update.", netConfig.patchPort, netConfig.ofPortPatch, curOfportPatch)
```

Comment:
Isn't this going to cause unnecessary logs on network startup? I think we should still check if the `curOfportPatch` is not empty.

Additionally, would it be possible to emit a warning event in this case? While I completely agree we should not exit on udn patch port missmatch I think an event would be a god way to highlight an important issue.

Reply:
A couple lines above there is a check 
```
if netConfig.ofPortPatch == "" {
			continue
		}
```
so it won't do anything until netconfig is updated with the patch port, the check with the empty port was only allowing port mismatch when the actual port is removed but the netConfig is not. I want to allow an actual mismatch when ofPort is newly configured, but the netConfig hasn't been updated yet.
I think an event I can try

Reply:
yeah you have to dig though to figure this out :D
so the netconfig ofport is set by gateway initFunc, and openflow_manager check_ports is started to gateway.Start, which happens after

Reply:
I have also tried adding an event, but that requires passing a lot of info, like event recorder and at least nodename (if we record event on the node or UDN if we record an event for UDN).
I am working on refactoring this part right now (first part here https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5345), so I will try to add an event recorder later when it is easier if that is ok